be/src/vec/functions/function_struct_in.h

```diff
@@ -0,0 +1,164 @@
+#include "vec/data_types/data_type_factory.hpp"
+#include "vec/data_types/data_type_nullable.h"
+#include "vec/data_types/data_type_number.h"
+#include "vec/functions/function.h"
+
+namespace doris::vectorized {
+struct ColumnRowRef {
+    ColumnPtr column;
+    size_t row_idx;
+
+    // equals when call set insert, this operator will be used
+    bool operator==(const ColumnRowRef& other) const {
+        return column->compare_at(row_idx, other.row_idx, *column, 0) == 0;
+    }
+    // compare
+    bool operator<(const ColumnRowRef& other) const {
+        return column->compare_at(row_idx, other.row_idx, *column, 0) < 0;
+    }
+
+    // when call set find, will use hash to find
+    size_t operator()(const ColumnRowRef& a) const {
+        std::vector<uint32_t> hash_vec(1);
+        auto* __restrict hash_val = hash_vec.data();
+        a.column->update_crc_with_value(a.row_idx, a.row_idx + 1, *hash_val, nullptr);
+        return *hash_val;
+    }
+};
+
+template <bool negative>
+class FunctionStructIn : public IFunction {
+public:
+    static constexpr auto name = negative ? "struct_not_in" : "struct_in";
+
+    static FunctionPtr create() { return std::make_shared<FunctionStructIn>(); }
+
+    String get_name() const override { return name; }
+
+    bool is_variadic() const override { return true; }
+
+    size_t get_number_of_arguments() const override { return 0; }
+
+    DataTypePtr get_return_type_impl(const DataTypes& args) const override {
+        for (const auto& arg : args) {
+            if (arg->is_nullable()) {
+                return make_nullable(std::make_shared<DataTypeUInt8>());
+            }
+        }
+        return std::make_shared<DataTypeUInt8>();
+    }
+
+    bool use_default_implementation_for_nulls() const override { return false; }
+
+    // make data in context into a set
+    Status open(FunctionContext* context, FunctionContext::FunctionStateScope scope) override {
+        DCHECK(context->get_num_args() >= 1);
+        auto* col_desc = context->get_arg_type(0);
+        DataTypePtr args_type = DataTypeFactory::instance().create_data_type(*col_desc);
+        MutableColumnPtr column_struct_ptr_args = remove_nullable(args_type)->create_column();
+        for (int i = 1; i < context->get_num_args(); ++i) {
+            // FE should make element type consistent and
+            // equalize the length of the elements in struct
+            const auto& const_column_ptr = context->get_constant_col(i);
+            if (const_column_ptr == nullptr) {
+                break;
+            }
+            const auto& [col, _] = unpack_if_const(const_column_ptr->column_ptr);
+            if (col->is_nullable()) {
+                auto* null_col = vectorized::check_and_get_column<vectorized::ColumnNullable>(col);
+                if (!null_in_set && null_col->has_null()) {
+                    null_in_set = true;
+                }
+                column_struct_ptr_args->insert_from(null_col->get_nested_column(), 0);
```

Comment:
if the value is NULL, no need call `insert_from`

Reply:
make sense! here need else